framework/generated/khronos_generators/khronos_struct_decoders_header_generator.py

```diff
@@ -59,6 +65,150 @@ def endFile(self):
             )
             write(body, file=self.outFile)
 
+        api_data = self.get_api_data()
+        var_name = api_data.type_prefix.lower() + '_type'
+        struct_type = api_data.struct_type_enum
+
+        for struct in self.get_all_filtered_struct_names():
```

Comment:
I think this section needs at least a comment.  Something like "Handle decoding children structs.  When a parent struct is encountered, at runtime we need to determine which actual child struct is referenced and allocate that struct instead of the parent struct."